src/main/java/org/kiwiproject/dropwizard/poller/metrics/DefaultClientPollerStatistics.java

```diff
@@ -0,0 +1,255 @@
+        avgPollLatency.set(cmaNew);
+    }
+
+    private static double cumulativeMovingAverage(double cmaOrig, long newValue, int numValues) {
+        return cmaOrig + ((newValue - cmaOrig) / (numValues + 1));
+    }
+
+    private void logSkippedMeasurement(long latencyInMillis) {
+        LOG.warn("Skipping measurement {} because lock was not obtained in {} {}",
+                latencyInMillis, latencyLocker.getLockWaitTime(), latencyLocker.getLockWaitTimeUnit());
+    }
+
+    @Override
+    public double averagePollLatencyInMillis() {
+        return avgPollLatency.get();
+    }
+
+    @Override
+    public void incrementCount() {
+        count.incrementAndGet();
+        lastAttemptTime.set(now());
+    }
+
+    @Override
+    public Optional<Long> lastAttemptTimeInMillis() {
+        long currentValue = lastAttemptTime.get();
+        return valueIfNonZeroOrElseEmpty(currentValue);
+    }
+
+    @Override
+    public void incrementSuccessCount() {
+        successCount.incrementAndGet();
+    }
+
+    @Override
+    public void incrementSkipCount() {
+        skipCount.incrementAndGet();
+        lastSkipTime.set(now());
+    }
+
+    @Override
+    public Optional<Long> lastSkipTimeInMillis() {
+        long currentValue = lastSkipTime.get();
+        return valueIfNonZeroOrElseEmpty(currentValue);
+    }
+
+    @Override
+    public void incrementFailureCount(String message) {
+        incrementFailureCount();
+        recentFailures.add(new FailedPollResult(message));
+    }
+
+    @Override
+    public void incrementFailureCount(Throwable exception) {
+        incrementFailureCount();
+        recentFailures.add(new FailedPollResult(exception));
+    }
+
+    private void incrementFailureCount() {
+        failureCount.incrementAndGet();
+        lastFailureTime.set(now());
+    }
+
+    private static long now() {
+        return System.currentTimeMillis();
+    }
+
+    @Override
+    public Optional<Long> lastFailureTimeInMillis() {
+        long currentValue = lastFailureTime.get();
+        return valueIfNonZeroOrElseEmpty(currentValue);
+    }
+
+    private static Optional<Long> valueIfNonZeroOrElseEmpty(long currentValue) {
+        if (currentValue == 0) {
+            return Optional.empty();
+        }
+        return Optional.of(currentValue);
+    }
+
+    @Override
+    public Stream<Long> recentFailureTimesInMillis() {
+        return recentFailures.stream().map(FailedPollResult::getTime);
+    }
+
+    @Override
+    public Stream<Map<String, Object>> recentFailureDetails() {
+        int skipFirstNElements = recentFailures.size() - numberOfFailureDetails();
+        return recentFailures.stream()
+                .skip(skipFirstNElements)
+                .map(FailedPollResult::getDetails);
+    }
+
+    @VisibleForTesting
+    int numberOfFailureDetails() {
+        return recentFailures.size() > FAILURE_DETAIL_LIMIT ? FAILURE_DETAIL_LIMIT : recentFailures.size();
```

Comment:
Suggest adding clarifying parentheses, e.g.

```suggestion
        return (recentFailures.size() > FAILURE_DETAIL_LIMIT) ? FAILURE_DETAIL_LIMIT : recentFailures.size();
```